### ISSUE TITLE
Lazy doesn't preserve contstructor.

### ISSUE THREAD

**Issue by jamesmontemagno** [mono/linker#1071]
On Xamarin.iOS we found this when we turn on the linker with the following code that the constructor will get linked out and will crash the app:

> The lazily-initialized type does not have a public, parameterless constructor.

```csharp
public static partial class TextToSpeech
    {
        static readonly Lazy<AVSpeechSynthesizer> speechSynthesizer = new Lazy<AVSpeechSynthesizer>();
 internal static async Task PlatformSpeakAsync(string text, SpeechOptions options, CancellationToken cancelToken = default)
        {
            using (var speechUtterance = GetSpeechUtterance(text, options))
            {
                await SpeakUtterance(speechUtterance, cancelToken);
            }
        }

        static AVSpeechUtterance GetSpeechUtterance(string text, SpeechOptions options)
        {
            var speechUtterance = new AVSpeechUtterance(text);

            if (options != null)
            {
                // null voice if fine - it is the default
                speechUtterance.Voice =
                    AVSpeechSynthesisVoice.FromLanguage(options.Locale?.Language) ??
                    AVSpeechSynthesisVoice.FromLanguage(AVSpeechSynthesisVoice.CurrentLanguageCode);

                // the platform has a range of 0.5 - 2.0
                // anything lower than 0.5 is set to 0.5
                if (options.Pitch.HasValue)
                    speechUtterance.PitchMultiplier = options.Pitch.Value;

                if (options.Volume.HasValue)
                    speechUtterance.Volume = options.Volume.Value;
            }

            return speechUtterance;
        }

        internal static async Task SpeakUtterance(AVSpeechUtterance speechUtterance, CancellationToken cancelToken)
        {
            var tcsUtterance = new TaskCompletionSource<bool>();
            try
            {
                speechSynthesizer.Value.DidFinishSpeechUtterance += OnFinishedSpeechUtterance;
                speechSynthesizer.Value.SpeakUtterance(speechUtterance);
                using (cancelToken.Register(TryCancel))
                {
                    await tcsUtterance.Task;
                }
            }
            finally
            {
                speechSynthesizer.Value.DidFinishSpeechUtterance -= OnFinishedSpeechUtterance;
            }

            void TryCancel()
            {
                speechSynthesizer.Value?.StopSpeaking(AVSpeechBoundary.Word);
                tcsUtterance?.TrySetResult(true);
            }

            void OnFinishedSpeechUtterance(object sender, AVSpeechSynthesizerUteranceEventArgs args)
            {
                if (speechUtterance == args.Utterance)
                    tcsUtterance?.TrySetResult(true);
            }
        }
}
```

From: https://github.com/xamarin/Essentials/issues/1207

**Comment by marek-safar** [comment]
This will be addressed by #988 for .NET5

/cc @vitek-karas

**Comment by marek-safar** [comment]
Closing as all the linker infrastructure for this landed in .NET 5.0
